autoload/lsp/handlers.vim

```diff
@@ -92,7 +92,9 @@ export def ProcessNotif(lspserver: dict<any>, reply: dict<any>): void
       'language/status': IgnoreNotif
     }
 
-  if lsp_notif_handlers->has_key(reply.method)
+  if lspserver.customNotificationHandlers->has_key(reply.method)
+    lspserver.customNotificationHandlers[reply.method](lspserver, reply)
+  elseif lsp_notif_handlers->has_key(reply.method)
```

Comment:
1. If a custom notification handler is provided it will ignore the built-ins Does that make sense?
2. Currently the internal API `(lspserver, reply)` is leaked. Does that make sense?

Reply:
1. If the built-in handler properly handles a notification (e.g. textDocument/publishDiagnostics), then the custom notification handler cannot override the built-in handler.  But if the built-in handler doesn't handle a notification or ignores it (e.g. language/status, $/typescriptVersion, etc.), then the custom handler can override it.
2. If the custom handler needs to send a message to the lsp server while processing the notification, then it will need "lspserver".  So passing "lspserver" and "reply" as parameter makes sense.